pkg/cmd/job/job_add.go

```diff
@@ -0,0 +1,72 @@
+package job
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-core-cli/pkg/metronome"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// newCmdClusterAdd creates a new job.
+func newCmdJobAdd(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add",
+		Short: "add a job",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := metronomeClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			// Handling input from file or stdin
+			var reader io.Reader
+			switch len(args) {
+			case 0:
+				if terminal.IsTerminal(int(os.Stdin.Fd())) {
+					return fmt.Errorf("input from the terminal is not accepted")
+				}
+				reader = os.Stdin
+				break
+			case 1:
+				reader, err = os.Open(args[0])
+				if err != nil {
+					return err
+				}
+				break
+			default:
+				return fmt.Errorf("input must be from stdin or file")
```

Comment:
This could use the Args field, eg. https://github.com/dcos/dcos-cli/blob/96375b35f1d379f092f23a5e4f677a0b73200c1e/pkg/cmd/auth/auth_listproviders.go#L19